### ISSUE TITLE
Query: Translate ToList.EnumerableMethod()

### ISSUE THREAD

**Issue by groege** [aspnet/EntityFrameworkCore#19466]
Given following query
```
_context.TransportOrder.Select(x => new TransportOrder() {
    TransportOrderData = new TransportOrderData() 
    {
        ExternalTransportOrderNumber = x.TransportOrderData.ExternalTransportOrderNumber, 
        StatusId = x.TransportOrderData.StatusId
    }, 
    TransportOrderSharedData = new TransportOrderSharedData() 
    {
        TransportOrderNumber = x.TransportOrderSharedData.TransportOrderNumber
    }, 
    Supplier = new TransportAddress() 
    {
        AddressName = x.Supplier.AddressName
    }, 
    Carrier = new TransportAddress() 
    {
        TechEmail = x.Carrier.TechEmail
    }, 
    Driver = new TransportAddress() {
        TechEmail = x.Driver.TechEmail
    }, 
    PickUpAddress = new TransportAddress()
    {
        AddressName = x.PickUpAddress.AddressName, 
        ZipCode = x.PickUpAddress.ZipCode, 
        City = x.PickUpAddress.City, 
        Street = x.PickUpAddress.Street
    }, 
    DeliveryAddress = new TransportAddress() {
        AddressName = x.DeliveryAddress.AddressName, 
        ZipCode = x.DeliveryAddress.ZipCode, 
        City = x.DeliveryAddress.City, 
        Street = x.DeliveryAddress.Street
    }, 
    SubCarrier = new TransportAddress() 
    {
        TechEmail = x.SubCarrier.TechEmail
    }, 
    LogisticsContactPersons = x.LogisticsContactPersons.Select(y => new TransportAddress() 
    {
         AddressName = y.Address.AddressName
    }).ToList(), 
    Projects = x.Projects.Select(y => new TransportProject() {
        ProjectNumber = y.Data.ProjectNumber, 
        Species = y.Data.Species, Assortment = y.Data.Assortment
}).ToList()})
//it doesn't matter whats inside any ==> y.ProjectNumber == null is broken as well
.Where(x => x.Projects.Any(y => EF.Functions.ILike(y.ProjectNumber, "%a%")))
.ToList();
```

Returns the following exception:

> System.InvalidOperationException
  HResult=0x80131509
  Message=The LINQ expression 'DbSet<Order>
    .Where(o => DbSet<OrderProject>
        .Where(o0 => EF.Property<Nullable<Guid>>(o, "TransportOrderGuid") != null && EF.Property<Nullable<Guid>>(o, "TransportOrderGuid") == EF.Property<Nullable<Guid>>(o0, "TransportOrderGuid"))
        .Select(o0 => new TransportProject{ 
            ProjectNumber = EF.Property<TransportProject>(o0, "Data").ProjectNumber, 
            Species = EF.Property<TransportProject>(o0, "Data").Species, 
            Assortment = EF.Property<TransportProject>(o0, "Data").Assortment 
        }
        )
        .ToList()
        .Any(y => __Functions_0
            .ILike(
                _: y.ProjectNumber, 
                matchExpression: "%a%")))' could not be translated. Either rewrite the query in a form that can be translated, or switch to client evaluation explicitly by inserting a call to either AsEnumerable(), AsAsyncEnumerable(), ToList(), or ToListAsync(). See https://go.microsoft.com/fwlink/?linkid=2101038 for more information.
  Source=Microsoft.EntityFrameworkCore
  StackTrace:
   at Microsoft.EntityFrameworkCore.Query.QueryableMethodTranslatingExpressionVisitor.<VisitMethodCall>g__CheckTranslated|8_0(ShapedQueryExpression translated, <>c__DisplayClass8_0& )
   at Microsoft.EntityFrameworkCore.Query.QueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
   at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor visitor)
   at Microsoft.EntityFrameworkCore.Query.QueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
   at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor visitor)
   at Microsoft.EntityFrameworkCore.Query.QueryCompilationContext.CreateQueryExecutor[TResult](Expression query)
   at Microsoft.EntityFrameworkCore.Storage.Database.CompileQuery[TResult](Expression query, Boolean async)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQueryCore[TResult](IDatabase database, Expression query, IModel model, Boolean async)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass12_0`1.<ExecuteAsync>b__0()
   at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler)
   at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQuery[TResult](Object cacheKey, Func`1 compiler)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.ExecuteAsync[TResult](Expression query, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.ExecuteAsync[TResult](Expression expression, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ExecuteAsync[TSource,TResult](MethodInfo operatorMethodInfo, IQueryable`1 source, Expression expression, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ExecuteAsync[TSource,TResult](MethodInfo operatorMethodInfo, IQueryable`1 source, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.CountAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at FreightExchange.Extensions.GraphQlExtensions.<GetResult>d__18`2.MoveNext() in C:\Users\haberl\source\repos\carriermobile\FreightExchange\Extensions\GraphQlExtensions.cs:line 401

### Further technical details

EF Core version: 3.1
Database provider: Npgsql.EntityFrameworkCore.PostgreSQL
Target framework: NET Core 3.1
Operating system: WIN 10
IDE: Visual Studio 2019 16.4.0 Preview 6.0

**Comment by ajcvickers** [comment]
@smitpatel to find dupe.

**Comment by smitpatel** [comment]
Fixed in #19377
